test/e2e/storage/testsuites/snapshotsecrets.go

```diff
@@ -0,0 +1,274 @@
+
+			// Depending on how far the test executed, cleanup accordingly
+			// Execute in reverse order, similar to defer stack
+			for i := len(cleanupSteps) - 1; i >= 0; i-- {
+				err := tryFunc(cleanupSteps[i])
+				framework.ExpectNoError(err, "while running cleanup steps")
+			}
+
+		}
+		ginkgo.BeforeEach(func() {
+			init()
+		})
+		ginkgo.AfterEach(func() {
+			cleanup()
+		})
+
+		ginkgo.Context("", func() {
+			var (
+				vs     *unstructured.Unstructured
+				secret *v1.Secret
+				sr     *SnapshotResource
+			)
+
+			ginkgo.BeforeEach(func() {
+
+				cleanupSteps = append(cleanupSteps, func() {
+					framework.ExpectNoError(sr.CleanupResource())
+				})
+				// create a secret
+				secret = &v1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: f.Namespace.Name,
+						Name:      "snapshot-secret",
+					},
+					Data: map[string][]byte{
+						"secret-data": []byte("secret-value-1"),
+					},
+				}
+
+				if secret, err = cs.CoreV1().Secrets(f.Namespace.Name).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+					framework.Failf("unable to create test secret %s: %v", secret.Name, err)
+				}
+
+				// Create snapshot using secret
+				parameters := map[string]string{
+					CSISnapshotterSecretName:      "snapshot-secret",
+					CSISnapshotterSecretNamespace: f.Namespace.Name,
+				}
+				sr = CreateSnapshotResource(sDriver, config, pattern, pvc.GetName(), pvc.GetNamespace(), parameters)
+				vs = sr.Vs
+			})
+			ginkgo.It("Use secrets field, create/list and delete snapshots", func() {
+
+				ginkgo.By("list the snapshot")
+				ListSnapshotsWithSecrets(dc, vs.GetNamespace(), sr, pattern)
+
+				ginkgo.By("Delete the snapshot")
+				DeleteSnapshotWithSecrets(cs, dc, vs.GetNamespace(), vs.GetName(), framework.Poll, framework.SnapshotDeleteTimeout)
+				ginkgo.By("checking if the snapshot is deleted")
+
+			})
+		})
+	})
+}
+
+// ListSnapshotsWithSecrets lists a given volumesnapshot it uses secret provided within storageclass
+func ListSnapshotsWithSecrets(dc dynamic.Interface, ns string, r *SnapshotResource, pattern testpatterns.TestPattern) (err error) {
+	ginkgo.By("listing the snapshot")
+	// ListSnapshot is called when you can create a VolumeSnapshot for pre-preprivisoned snapshots.
+	// So we exploit this to test ListSnapshot with secrets
+
+	ginkgo.By("recording the volume handle and snapshotHandle")
+	snapshotHandle := r.Vscontent.Object["status"].(map[string]interface{})["snapshotHandle"].(string)
+	framework.Logf("Recording snapshot handle: %s", snapshotHandle)
+	csiDriverName := r.Vsclass.Object["driver"].(string)
+
+	ginkgo.By("creating a snapshot content with the snapshot handle")
+	uuid := uuid.NewUUID()
+
+	snapName := getPreProvisionedSnapshotName(uuid)
+	snapcontentName := getPreProvisionedSnapshotContentName(uuid)
+
+	r.Vscontent = getPreProvisionedSnapshotContent(snapcontentName, snapName, ns, snapshotHandle, pattern.SnapshotDeletionPolicy.String(), csiDriverName)
```

Comment:
To test ListSnapshots, we need to add VolumeSnapshotClassName to the VolumeSnapshotContent for pre-provisoned case and make sure the parameters contain the list snapshots secrets.  So you'll need to modify getPreProvisionedSnapshotContent for this test.